airflow/www/static/js/api/useSetDagRunNotes.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import axios from 'axios';
+import { useMutation, useQueryClient } from 'react-query';
+
+import { getMetaValue } from 'src/utils';
+import type { API } from 'src/types';
+import useErrorToast from 'src/utils/useErrorToast';
+
+import { emptyGridData } from './useGridData';
+import type { GridData } from './useGridData';
+
+const setDagRunNotesURI = getMetaValue('set_dag_run_notes');
+
+interface Props {
+  dagId: string;
+  runId: string;
+}
```

Comment:
For late: Too bad we cannot use `SetDagRunNotesVariables` to type this. I think we should separate the type generation for `parameters/path & query` and `requetsBody`.

This way we could easily type such mutation, keeping type enforced by the OpenAPI spec. What do you think @bbovenzi

Reply:
For instance the `notes`  type would raise a type error I believe, it can be `undefined | string` but not `null | string` I believe.